ale_linters/java/javac.vim

```diff
@@ -0,0 +1,106 @@
+" Author: farenjihn <farenjihn@gmail.com>
+" Description: Lints java files using javac
+
+if exists('g:loaded_ale_linters_java_javac')
+    finish
+endif
+
+let g:loaded_ale_linters_java_javac = 1
+let g:ale_java_javac_classpath = ''
+
+let s:eclipse_classpath = ''
+let s:tmppath = '/tmp/java_ale/'
```

Comment:
That's the plan. We need to first write a PowerShell script to replace the executable for Windows, so it writes to a temporary directory too, and then replace the file extension arguments with just the filename itself. I'd suggest replacing all of the concatenation with `ale#util#stdin_wrapper` with a `wrap_stdin` argument for the linters.